governance/program/src/processor/process_insert_instruction.rs

```diff
@@ -0,0 +1,106 @@
+//! Program state processor
+
+use std::cmp::Ordering;
+
+use borsh::BorshSerialize;
+use solana_program::{
+    account_info::{next_account_info, AccountInfo},
+    entrypoint::ProgramResult,
+    pubkey::Pubkey,
+    rent::Rent,
+    sysvar::Sysvar,
+};
+
+use crate::{
+    error::GovernanceError,
+    state::{
+        enums::GovernanceAccountType,
+        governance::get_governance_data,
+        proposal::get_proposal_data_for_governance,
+        proposal_instruction::{
+            get_proposal_instruction_address_seeds, InstructionData, ProposalInstruction,
+        },
+        token_owner_record::get_token_owner_record_data_for_proposal_owner,
+    },
+    tools::account::create_and_serialize_account_signed,
+};
+
+/// Processes InsertInstruction instruction
+pub fn process_insert_instruction(
+    program_id: &Pubkey,
+    accounts: &[AccountInfo],
+    index: u16,
+    hold_up_time: u64,
+    instruction: InstructionData,
+) -> ProgramResult {
+    let account_info_iter = &mut accounts.iter();
+
+    let governance_info = next_account_info(account_info_iter)?; // 0
+    let proposal_info = next_account_info(account_info_iter)?; // 1
+    let token_owner_record_info = next_account_info(account_info_iter)?; // 2
+    let governance_authority_info = next_account_info(account_info_iter)?; // 3
+
+    let proposal_instruction_info = next_account_info(account_info_iter)?; // 4
+
+    let payer_info = next_account_info(account_info_iter)?; // 5
+    let system_info = next_account_info(account_info_iter)?; // 6
+
+    let rent_sysvar_info = next_account_info(account_info_iter)?; // 7
+    let rent = &Rent::from_account_info(rent_sysvar_info)?;
+
+    if !proposal_instruction_info.data_is_empty() {
+        return Err(GovernanceError::InstructionAlreadyExists.into());
+    }
+
+    let governance_data = get_governance_data(governance_info)?;
+
+    if hold_up_time < governance_data.config.min_instruction_hold_up_time {
+        return Err(GovernanceError::InstructionHoldUpTimeBelowRequiredMin.into());
+    }
+
+    let mut proposal_data = get_proposal_data_for_governance(&proposal_info, governance_info.key)?;
+    proposal_data.assert_can_edit_instructions()?;
+
+    let token_owner_record_data = get_token_owner_record_data_for_proposal_owner(
```

Comment:
Agree, only `Proposal` owner (the user who created it) can edit it, including editing instructions. Once `Proposal` is completed the owner can request `Signatories` to sign off the `Proposal` and submit it for voting.
Note, if the owner specified `Governance Delegate` then either the `Owner` or the `Delegate` can create and edit `Proposals` because they have the same permissions within the `Realm`.